### ISSUE TITLE
Do a Barrel Roll Crashing Fabric

### ISSUE THREAD

**Issue by bettskyrin** [enjarai/do-a-barrel-roll#96]
**Describe the issue**
Minecraft crashes on launch.

**Crash report/logs (if applicable)**
Curseforge Modloader - Fabric version 0.14.22
Minecraft version 1.20.1
Do a Barrel Roll version 3.1.0
Exit Code 1

**Comment by 3quarterscolon** [comment]
AFAIK exit code 1 just means the application had an error and doesn't explain anything specific. you should probably look for the generated crash report so the creator of the mod can diagnose the issue.

**Comment by iblowmymind** [comment]
A mixin errror is thrown with Do a Barrel Roll 3.1.0 on both Fabric/Quilt client and server.
Trace:
```
org.quiltmc.loader.impl.FormattedException: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_1309 failed
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:526)
	at org.quiltmc.loader.impl.launch.knot.Knot.launch(Knot.java:82)
	at org.quiltmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:28)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at needleWrapper.dev.kosmx.needle.launchWrapper.ParameterizedWrapper.main(ParameterizedWrapper.kt:12)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71)
Caused by: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_1309 failed
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:458)
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:244)
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.loadClassOnly(KnotClassDelegate.java:164)
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:150)
	at org.quiltmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:228)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1013)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150)
	at org.quiltmc.loader.impl.launch.knot.KnotClassLoader.defineClassFwd(KnotClassLoader.java:310)
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:299)
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.loadClassOnly(KnotClassDelegate.java:164)
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:150)
	at org.quiltmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:228)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	at net.minecraft.class_2966.method_12851(class_2966.java:54)
	at net.minecraft.client.main.Main.main(Main.java:166)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:524)
	... 8 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202)
	at org.quiltmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:453)
	... 27 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Variable modifier method doABarrelRoll$modifyKineticDamage(F)F in #do_a_barrel_roll:do-a-barrel-roll.mixins.json:LivingEntityMixin from mod do_a_barrel_roll failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap do-a-barrel-roll-refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:468)
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1384)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1062)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:402)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:327)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
	... 30 more
```

**Comment by enjarai** [comment]
Ah fuck, made a mistake there. I'll need some time to fix this, so use the previous version for now.

**Comment by iblowmymind** [comment]
Alright, please update us when the fix is pushed!
Thank you!

**Comment by enjarai** [comment]
Should™ be fixed in 3.1.1.
